### PR TITLE
PEtab: More informative output for unhandled species overrides

### LINKED ISSUE #1516: PEtab handling fails for problems with species in the condition table that have initialAmount and hasOnlySubstanceUnits=False
PEtab handling fails for problems with species in the condition table that have initialAmount and hasOnlySubstanceUnits=False, specifically in `amici.petab_objective.create_parameter_mapping_for_condition`.

The problem is, that here a float is expected, but a sympy.Expr occurrs:

https://github.com/AMICI-dev/AMICI/blob/c0377065207b68c881143d203de1f56fd52878b9/python/amici/petab_objective.py#L348-L353

Can be fixed for constant volumes. Variable volumes cannot currently be handled.

### PR DESCRIPTION
**Pull request by dweindl**
Handle sympy expressions that consist only of parameter IDs.
Otherwise show the offending expression.

Partially resolves #1516

### CODE REVIEW

**Review by dilpath — APPROVED** [review]
:+1:

File: python/amici/petab_objective.py [hunk 1]
```diff
@@ -346,11 +347,21 @@ def _set_initial_concentration(condition_id, species_id, init_par_id,
...
+                    value = float(value)
+                except ValueError:
+                    if sp.nsimplify(value).is_Atom:
+                        # Get rid of multiplication with one
+                        value = sp.nsimplify(value)
+                    else:
+                        raise NotImplementedError(
+                            "Cannot handle non-trivial expressions for "
+                            f"species initial for {species_id}: {value}")
+                    # this should be a parameter ID
```
[the remark this answers is not in the document]
Comment (dweindl) [reply python/amici/petab_objective.py:363]: Could be a parameter of any kind.

[the remark this answers is not in the document]
Comment (dilpath) [reply python/amici/petab_objective.py:363]: Would this make sensitivity equations condition-dependent? Is this supported?

[the remark this answers is not in the document]
Comment (dweindl) [reply python/amici/petab_objective.py:363]: This is about mapping PEtab problem-parameters to AMICI model parameters. The sensitivities are computed with respect to the model parameter, which is then mapped back to the PEtab problem parameter. Unless I am not completely overlooking something, this should work. But we definitely need to extend the PEtab test suite.

File: python/amici/petab_objective.py [hunk 2]
```diff
@@ -346,11 +347,21 @@ def _set_initial_concentration(condition_id, species_id, init_par_id,
...
+                    value = str(value)
```
Comment (FFroehlich) [inline python/amici/petab_objective.py:364]: don't we only want to do this if value is not numeric?

Comment (FFroehlich) [reply python/amici/petab_objective.py:364]: ah! comments made this a little hard to spot ;)

**Pull request merged**